test/unit/specs/components/Personalization/createOnClickHandler.spec.js

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+import createOnClickHandler from "../../../../../src/components/Personalization/createOnClickHandler";
+
+describe("Personalization::createOnClickHandler", () => {
+  let mergeMeta;
+  let collectClicks;
+  const event = { type: "blah" };
+  const clickStorage = {};
+
+  beforeEach(() => {
+    mergeMeta = jasmine.createSpy("mergeMeta");
+    collectClicks = jasmine.createSpy("collectClicks");
+  });
+
+  it("collects clicks", () => {
+    const handleOnClick = createOnClickHandler({
+      mergeMeta,
+      collectClicks,
+      clickStorage
+    });
+    const clickedElement = "foo";
+    handleOnClick({ event, clickedElement });
+    expect(collectClicks).toHaveBeenCalledWith(
+      jasmine.any(Function),
```

Comment:
Because this function was created by the module, I recommend adding a test to make sure that the function does what you expect it to do. Specifically, it should call `mergeMeta` with `event` and `meta` and return the result.

Reply:
there is a proposal to change the clickTracking functionality, and I would rather add these changes then, if it will still apply.